spec/requests/providers/statement_of_case_spec.rb

```diff
@@ -45,18 +46,18 @@
   end
 
   describe 'PATCH providers/proceedings_before_the_court' do
-    subject { patch providers_legal_aid_application_statement_of_case_path(legal_aid_application), params: params.merge(submit_button).merge(upload_button) }
     let(:entered_text) { Faker::Lorem.paragraph(3) }
-    let(:original_file) { uploaded_file('spec/fixtures/files/lorem_ipsum.pdf', 'application/pdf') }
-
-    let(:params) do
+    let(:original_file) { uploaded_file('spec/fixtures/files/documents/hello_world.pdf', 'application/pdf') }
+    let(:statement_of_case) { legal_aid_application.statement_of_case }
+    let(:params_statement_of_case) do
       {
-        statement_of_case: {
-          statement: entered_text,
-          original_files: [original_file]
-        }
+        statement: entered_text,
+        original_files: [original_file]
       }
     end
+    let(:params) { { statement_of_case: params_statement_of_case }.merge(submit_button).merge(upload_button) }
```

Comment:
As I don't think there is a scenario where the two button are pressed together ... rather than `.merge(submit_button).merge(upload_button)` I'd suggest:
```rspec
let(:draft_button) { { draft_button: 'Save as draft' } }
let(:upload_button) { { upload_button: 'Upload' } }
let(:button_clicked) { { } }
let(:params) { { statement_of_case: params_statement_of_case }.merge(button_clicked) }
```
Then in the spec where draft button behaviour is tested:
```rspec
let(:button_clicked) { draft_button }
```
And where the upload button is used:
```rspec
let(:button_clicked) { upload_button }
```